datacube/utils/changes.py

```diff
@@ -6,9 +6,15 @@
 Validation of document/dictionary changes.
 """
 from itertools import zip_longest
+from typing import Any, Callable, List, Mapping, Tuple, Union
 
+# Type that can be checked for changes.
+# (MyPy approximation without recursive references)
+Changable = Union[str, int, None, List[Any], Mapping[str, Any]]
```

Comment:
Does it really expect a `List` and not more generic `Sequence[Any]`? Is input never a `Tuple[Any,...]` for example?

Reply:
Sequence would probably be a better choice, yes.